File: design-patterns/go/decorator/ui_element/margin_decorator.go
// margin_decorator.go

package main

import "fmt"

type MarginDecorator struct {
	*UIDecorator
}

func NewMarginDecorator(uiElement UIElement) (marginDecorator *MarginDecorator) {
	marginDecorator = &MarginDecorator{}
	marginDecorator.UIDecorator = NewUIDecorator(uiElement)
	return
}

func (marginDecorator *MarginDecorator) Draw() {
	// Can perform any additional task anywhere in the method

	marginDecorator.UIDecorator.Draw()

	// Write code to add margin to the element
	fmt.Println("Adding margin to the element")
}

File: design-patterns/php/decorator/uielement/MarginDecorator.php
<?php
// MarginDecorator.php

namespace BigBoxCode\DesignPattern\Decorator\UiElement;

class MarginDecorator extends UIDecorator {
    public function __construct(UIElement $uiElement) {
        parent::__construct($uiElement);
    }

    public function draw(): void {
        parent::draw();
        
        echo "Adding margin to the element\n";
    }
}